pennylane/devices/preprocess.py

```diff
@@ -154,7 +154,7 @@ def mid_circuit_measurements(
     """
 
     if tape.shots:
-        return qml.dynamic_one_shot(tape)
+        return (tape,), null_postprocessing
```

Comment:
Heads up that this transform will change when #5679 gets merged.

Reply:
One more reason to get familiar with it :)